pycbc/inference/sampler/cpnest.py

```diff
@@ -0,0 +1,219 @@
+
+import numpy, os
+import cpnest
+import logging
+import  cpnest.model as cpm
+#from pycbc.pool import choose_pool
+
+from .base import BaseSampler
+from .base_mcmc import (BaseMCMC, raw_samples_to_dict,
+                        get_optional_arg_from_config)
+
+from pycbc.inference.io import (CPNestFile,validate_checkpoint_files)
+from .. import models
+
+
+#
+# =============================================================================
+#
+#                                   Samplers
+#
+# =============================================================================
+#
+
+class CPNestSampler(BaseSampler):
+    """This class is used to construct an NestedSampling sampler from the emcee
+    package's EnsembleSampler.
+
+    Parameters
+    ----------
+    model : model
+        A model from ``pycbc.inference.models``.
+    nlive : int
+        Number of walkers to use in sampler.
+    pool : function with map, Optional
+        A provider of a map function that allows a function call to be run
+        over multiple sets of arguments and possibly maps them to
+        cores/nodes/etc.
+    """
+    name = "cpnest"
+    _io = CPNestFile
+
+    #burn_in_class = MCMCBurnInTests
+
+    def __init__(self, model, nlive, checkpoint_interval=None,
+                 logpost_function=None, nthreads=8, maxmcmc=1000,resume=True):
+
+        self.model = model
+        self.nlive = nlive
+        self.maxmcmc = maxmcmc
+        self.nthreads = nthreads
+        #self.names = model.sampling_params
+        # create a wrapper for calling the model
+        if logpost_function is None:
+            logpost_function = 'logposterior'
+        self.model_call = CPnestModel(model)
+        self._sampler = None
+        #self._samples = None
+        self._nested_samples = None
+        self._posterior_samples = None
+        self._logz = None
+        self._dlogz = None
+        self.checkpoint_file = 'test.out' # !!!!!!!!!FIXME!!!!!!!!!!
+    
+    def run(self, **kwargs):
+        OUTDIR = os.path.dirname(os.path.abspath(self.checkpoint_file))
+        if self._sampler is None:
+            self._sampler = cpnest.CPNest(self.model_call,verbose=1,output=OUTDIR,nthreads=self.nthreads,nlive=self.nlive,maxmcmc=self.maxmcmc,resume=True)
+        res = self._sampler.run()
+
+    @property
+    def io(self):
+        return self._io
+
+    @property
+    def niterations(self):
+        return len(self.samples.values()[0])
+
+    @classmethod
+    def from_config(cls, cp, model,nprocesses=1,use_mpi=False):
+        """
+        Loads the sampler from the given config file.
+        """
+        section = "sampler"
+        # check name
+        assert cp.get(section, "name") == cls.name, (
+            "name in section [sampler] must match mine")
+        # get the number of live points to use
+        nlive = int(cp.get(section, "nlive"))
+        maxmcmc = int(cp.get(section, "maxmcmc"))
+        nthreads = int(cp.get(section, "nthreads"))
```

Comment:
Add something here to get the verbosity level, which you will then pass to the class initialization below.